File: arrays-and-hashing/01-contains-duplicate/01-contains-duplicate.py
# Using hashset
# time complexity: O(n)
# space complexity: O(n)
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        hashSet = set()

        for n in nums:
            if n in hashSet:
                return True
            hashSet.add(n)
        return False


s1 = Solution()
print(s1.containsDuplicate([1, 2, 3, 1]))

File: arrays-and-hashing/01-contains-duplicate/01-contains-duplicate.js
/**
 * Using hashset
 * time complexity:
 * space complexity:
 * @param {number[]} nums
 * @return {boolean}
 */
let containsDuplicate = function (nums) {
  const uniqueNums = new Set(nums);
	return uniqueNums.size < nums.length ? true : false;
};

console.log(containsDuplicate([1,2,3,1]));
console.log(containsDuplicate([1,2,3,4]));
console.log(containsDuplicate([1,1,1,3,3,4,3,2,4,2]));
